cli/deploy-profiling-job.sh

```diff
@@ -0,0 +1,137 @@
+else
+  echo "Compute $COMPUTE_NAME exists, skip creation."
+fi
+# </create_compute_cluster_for_hosting_the_profiler>
+
+# <create_endpoint>
+az ml endpoint show --name $ENDPOINT_NAME
+if [[ $? -ne 0 ]]; then
+  echo "Creating Endpoint $ENDPOINT_NAME ..."
+  az ml endpoint create --name $ENDPOINT_NAME -f endpoints/online/managed/simple-flow/1-create-endpoint-with-blue.yml
+else
+  echo "Endpoint $ENDPOINT_NAME exists, skip creation."
+fi
+# </create_endpoint>
+
+# <check_endpoint_Status>
+endpoint_status=`az ml endpoint show --name $ENDPOINT_NAME --query "provisioning_state" -o tsv`
+echo $endpoint_status
+if [[ $endpoint_status == "Succeeded" ]]; then
+  echo "Endpoint $ENDPOINT_NAME created successfully"
+else 
+  echo "Endpoint $ENDPOINT_NAME creation failed"
+  exit 1
+fi
+
+deploy_status=`az ml endpoint show --name $ENDPOINT_NAME --query "deployments[?name=='$DEPLOYMENT_NAME'].provisioning_state" -o tsv`
+echo $deploy_status
+if [[ $deploy_status == "Succeeded" ]]; then
+  echo "Deployment $DEPLOYMENT_NAME completed successfully"
+else
+  echo "Deployment $DEPLOYMENT_NAME failed"
+  exit 1
+fi
+# </check_endpoint_Status>
+
+# <create_profiling_job_yaml_file>
+# please specify environment variable "IDENTITY_ACCESS_TOKEN" when working with ml compute with no appropriate MSI attached
+sed \
+  -e "s/<% ENDPOINT_NAME %>/$ENDPOINT_NAME/g" \
+  -e "s/<% DEPLOYMENT_NAME %>/$DEPLOYMENT_NAME/g" \
+  -e "s/<% PROFILING_TOOL %>/$PROFILING_TOOL/g" \
+  -e "s/<% DURATION %>/$DURATION/g" \
+  -e "s/<% CONNECTIONS %>/$CONNECTIONS/g" \
+  -e "s/<% TARGET_RPS %>/$TARGET_RPS/g" \
+  -e "s/<% CLIENTS %>/$CLIENTS/g" \
+  -e "s/<% TIMEOUT %>/$TIMEOUT/g" \
+  -e "s/<% THREAD %>/$THREAD/g" \
+  -e "s/<% COMPUTE_NAME %>/$COMPUTE_NAME/g" \
+  endpoints/online/profiling/profiling_job_tmpl.yml > profiling_job.yml
+# </create_profiling_job_yaml_file>
+
+# <create_profiling_job>
+run_id=$(az ml job create -f profiling_job.yml --query name -o tsv)
+# </create_profiling_job>
+
+# <check_job_status_in_studio>
+az ml job show -n $run_id --web
+# </check_job_status_in_studio>
+
+# <stream_job_logs_to_console>
+az ml job stream -n $run_id
+sleep 10
+# </stream_job_logs_to_console>
+
+# <get_job_report>
+# get output datastore info
+output=$(az ml job show -n $run_id --query 'output' -o tsv)
+output_arr=($output)
+datastore_id=$(echo ${output_arr[0]} | awk -F: '{print $NF}')
+output_path=${output_arr[1]}
+
+# get storage info
+storage_info=$(az ml datastore show --include-secrets --name $datastore_id --query '[account_name,container_name,credential.access_key]' -o tsv)
+storage_info_arr=($storage_info)
+storage_account_name=${storage_info_arr[0]}
+storage_container_name=${storage_info_arr[1]}
+storage_key=${storage_info_arr[2]}
+
+# download job report
+az storage blob download --container-name $storage_container_name/$output_path/outputs --name report.json --file report_$run_id.json --account-name $storage_account_name --account-key $storage_key
```

Comment:
have you tried az ml job download? this command is supposed to download all job outputs including logs. it might be only in private cli tho.

Reply:
sure, I'll use az job download to replace the code above